src/librustc/traits/mod.rs

```diff
@@ -826,17 +829,23 @@ impl<'tcx,O> Obligation<'tcx,O> {
 impl<'tcx> ObligationCause<'tcx> {
     pub fn new(span: Span,
                body_id: ast::NodeId,
+               node_id: ast::NodeId,
```

Comment:
I haven't found any call to `new` that didn't use the same thing for `body_id` and `node_id`. Did I miss something?

Reply:
For now, I changed the code so that I could later fill `node_id` with the appropriate node (so that I wouldn't have to iterate the snippet).

Reply:
By the way, are we going to leave these changes (adding `node_id` to `ObligationCause`) even if we are not really using it (yet)?

Reply:
Given it'd be dead code, lets remove it and land this PR. We can make that change again once the `node_id` is actually used.